resources/assets/js/annotations/components/annotationCanvas/polygonBrushInteraction.vue

```diff
@@ -100,33 +100,44 @@ export default {
             }
         },
     },
-    created() {
-        if (this.canAdd) {
-            Keyboard.on('e', this.togglePolygonBrush, 0, this.listenerSet);
-        }
-
-        if (this.canModify) {
-            Keyboard.on('r', this.togglePolygonEraser, 0, this.listenerSet);
-            Keyboard.on('t', this.togglePolygonFill, 0, this.listenerSet);
-        }
-
-        if (this.canAdd || this.canModify) {
-            this.$watch('interactionMode', this.toggleCurrentInteraction);
+    watch: {
+        interactionMode(mode) {
+            if (this.canAdd || this.canModify) {
+                this.toggleCurrentInteraction(mode);
+            }
+        },
+        canAdd(state) {
+            if (state) {
+                Keyboard.on('e', this.togglePolygonBrush, 0, this.listenerSet);
+            }
+            else {
+                Keyboard.off('e', this.togglePolygonBrush, 0, this.listenerSet);
+            }
+        },
+        canModify(state) {
+            if (state) {
+                Keyboard.on('r', this.togglePolygonEraser, 0, this.listenerSet);
+                Keyboard.on('t', this.togglePolygonFill, 0, this.listenerSet);
+            }
+            else {
+                shiftClickSelectInteraction.setActive(false);
```

Comment:
This could lead to an invalid state (see my review comment).

I think you can already use `resetInteractionMode()` here because the `toggle*` methods already handle `can*`.